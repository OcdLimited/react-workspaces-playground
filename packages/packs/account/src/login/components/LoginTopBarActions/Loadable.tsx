import React from 'react';
import loadable from '@loadable/component';
import { Loading } from '@ocdlimited/abp.react.core';

const fallback = (
	<div>
		<Loading color="secondary" className="loading-content" />
	</div>
);

export const LoginTopBarActions = loadable(() => import('./LoginTopBarActions'), {
	fallback,
});
