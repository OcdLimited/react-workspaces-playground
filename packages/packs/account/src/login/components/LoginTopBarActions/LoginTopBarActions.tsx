import React from 'react';
import { LanguageSwitcher, LanguageSwitcherProps, SettingsCog } from '@ocdlimited/abp.react.theme.shared';

interface LoginTopBarActionsProps extends LanguageSwitcherProps {}

export function LoginTopBarActions({ onLanguageChange, languages, currentCulture }: LoginTopBarActionsProps) {
	return (
		<React.Fragment>
			<LanguageSwitcher
				onLanguageChange={onLanguageChange}
				languages={languages}
				currentCulture={currentCulture}
			/>
			<SettingsCog />
		</React.Fragment>
	);
}

export default LoginTopBarActions;
