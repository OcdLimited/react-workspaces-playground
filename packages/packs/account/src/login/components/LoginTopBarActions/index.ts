export { default as LoginTopBarActions } from './LoginTopBarActions';
