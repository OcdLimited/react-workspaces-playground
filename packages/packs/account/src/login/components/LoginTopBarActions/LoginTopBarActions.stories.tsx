import React from 'react';
import { LoginTopBarActions } from './LoginTopBarActions';

export default {
	title: 'account/Components/LoginTopBarActions',
};

export const Default = () => <LoginTopBarActions />;

Default.story = {
	name: 'Default',
};
