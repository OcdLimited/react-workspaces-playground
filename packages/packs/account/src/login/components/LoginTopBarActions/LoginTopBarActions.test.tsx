import React from 'react';
import { render } from '@testing-library/react';
import { LoginTopBarActions } from './LoginTopBarActions';

it('should render', () => {
	const { container } = render(
		<LoginTopBarActions
			onSelectLanguage={jest.fn()}
			languages={[]}
			currentCulture={{}}
			currentTheme=""
			themes={[]}
		/>,
	);

	expect(container).toMatchSnapshot();
});
