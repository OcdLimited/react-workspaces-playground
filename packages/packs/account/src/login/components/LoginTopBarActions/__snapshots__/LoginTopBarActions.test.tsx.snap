// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div>
  <button
    aria-haspopup="true"
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
    data-testid="open-language"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      <span
        class="MuiBadge-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm6.93 6h-2.95c-.32-1.25-.78-2.45-1.38-3.56 1.84.63 3.37 1.91 4.33 3.56zM12 4.04c.83 1.2 1.48 2.53 1.91 3.96h-3.82c.43-1.43 1.08-2.76 1.91-3.96zM4.26 14C4.1 13.36 4 12.69 4 12s.1-1.36.26-2h3.38c-.08.66-.14 1.32-.14 2 0 .68.06 1.34.14 2H4.26zm.82 2h2.95c.32 1.25.78 2.45 1.38 3.56-1.84-.63-3.37-1.9-4.33-3.56zm2.95-8H5.08c.96-1.66 2.49-2.93 4.33-3.56C8.81 5.55 8.35 6.75 8.03 8zM12 19.96c-.83-1.2-1.48-2.53-1.91-3.96h3.82c-.43 1.43-1.08 2.76-1.91 3.96zM14.34 14H9.66c-.09-.66-.16-1.32-.16-2 0-.68.07-1.35.16-2h4.68c.09.65.16 1.32.16 2 0 .68-.07 1.34-.16 2zm.25 5.56c.6-1.11 1.06-2.31 1.38-3.56h2.95c-.96 1.65-2.49 2.93-4.33 3.56zM16.36 14c.08-.66.14-1.32.14-2 0-.68-.06-1.34-.14-2h3.38c.16.64.26 1.31.26 2s-.1 1.36-.26 2h-3.38z"
          />
        </svg>
        <span
          class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary MuiBadge-invisible"
        />
      </span>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <button
    aria-haspopup="true"
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
    data-testid="open-language"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      <span
        class="MuiBadge-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.95 10.78c.03-.25.05-.51.05-.78s-.02-.53-.06-.78l1.69-1.32c.15-.12.19-.34.1-.51l-1.6-2.77c-.1-.18-.31-.24-.49-.18l-1.99.8c-.42-.32-.86-.58-1.35-.78L12 2.34c-.03-.2-.2-.34-.4-.34H8.4c-.2 0-.36.14-.39.34l-.3 2.12c-.49.2-.94.47-1.35.78l-1.99-.8c-.18-.07-.39 0-.49.18l-1.6 2.77c-.1.18-.06.39.1.51l1.69 1.32c-.04.25-.07.52-.07.78s.02.53.06.78L2.37 12.1c-.15.12-.19.34-.1.51l1.6 2.77c.1.18.31.24.49.18l1.99-.8c.42.32.86.58 1.35.78l.3 2.12c.04.2.2.34.4.34h3.2c.2 0 .37-.14.39-.34l.3-2.12c.49-.2.94-.47 1.35-.78l1.99.8c.18.07.39 0 .49-.18l1.6-2.77c.1-.18.06-.39-.1-.51l-1.67-1.32zM10 13c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3z"
            transform="scale(1.2, 1.2)"
          />
        </svg>
        <span
          class="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary MuiBadge-invisible"
        />
      </span>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
